src/core/plot/qgsvectorlayerplotdatagatherer.h

```diff
@@ -0,0 +1,124 @@
+/***************************************************************************
+                         qgsvectorlayerplotdatagatherer.h
+                         -----------------
+    begin                : August 2025
+    copyright            : (C) 2025 by Mathieu
+    email                : mathieu at opengis dot ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSVECTORLAYERPLOTDATAGATHERER_H
+#define QGSVECTORLAYERPLOTDATAGATHERER_H
+
+#include "qgis.h"
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsplot.h"
+#include "qgsvectorlayer.h"
+#include <qgsvectorlayerfeatureiterator.h>
+
+#include <QThread>
+
+
+/**
+ * \ingroup core
+ * \class QgsVectorLayerAbstractPlotDataGatherer
+ *
+ * \brief An abstract vector layer plot data gatherer base class.
+ *
+ * \since QGIS 4.0
+ */
+class CORE_EXPORT QgsVectorLayerAbstractPlotDataGatherer : public QThread
+{
+    Q_OBJECT
+
+  public:
+
+    QgsVectorLayerAbstractPlotDataGatherer() = default;
+    virtual ~QgsVectorLayerAbstractPlotDataGatherer() = default;
+
+  private:
+
+};
+
+
+/**
+ * \ingroup core
+ * \class QgsVectorLayerXyPlotDataGatherer
+ *
+ * \brief An vector layer plot data gatherer class for XY series.
+ *
+ * \since QGIS 4.0
+ */
+class CORE_EXPORT QgsVectorLayerXyPlotDataGatherer : public QgsVectorLayerAbstractPlotDataGatherer
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * XY series details
+     */
+    struct XySeriesDetails
+    {
+      explicit XySeriesDetails( const QString &xExpression, const QString &yExpression, const QString &orderByExpression = QString(), const QString &filterExpression = QString() )
+        : xExpression( xExpression )
+        , yExpression( yExpression )
+        , orderByExpression( orderByExpression )
+        , filterExpression( filterExpression )
+      {}
+
+      QString xExpression;
+      QString yExpression;
+      QString orderByExpression;
+      QString filterExpression;
```

Comment:
Instead of using arguments for these, how about changing the QgsVectorLayer argument to instead accept a feature iterator? That'd mean the caller would have full control over all the filtering/sorting options available via QgsFeatureRequest (and also allow this class to be used with non vector layer based feature iterators)